### PR TITLE
database sql tracer

### PR DESCRIPTION
**Pull request by bendiknesbo**
- feat: Add RegisterDriverAndOpen to trace "database/sql"-queries

Closes part of #180

### CODE REVIEW

**Review by stephendrake — APPROVED** [review]

**Comment by stephendrake** [comment]
I think something went wrong with the merge/rebase as the old MySQL specific files are back.

File: middleware/mysql/mysql.go
[the remark this answers is not in the document]
Comment (bendiknesbo) [reply middleware/mysql/mysql.go:1]: Thanks, fixed now

**Review by stephendrake — APPROVED** [review]

File: middleware/database/database.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	sqltrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql"
+)
+
+// RegisterDriverAndOpen registers the selected driver with the datadog-lib, and opens a connection to the database using the dsn.
+func RegisterDriverAndOpen(driverName string, driver driver.Driver, dsn string, options ...Option) (*sql.DB, error) {
+	cfg := defaults()
+	for _, opt := range options {
+		opt(cfg)
+	}
+	opts := make([]sqltrace.Option, 0, 4)
```
Comment (arunpoudel) [inline middleware/database/database.go:17]: If you are trying to fix the capacity and trying to minimize re-allocations, should it not be?

```
opts := make([]sqltrace.Option, 0, 2 + len(cfg.tags) + len(ignoredQueryTypes))
```

Comment (arunpoudel) [reply middleware/database/database.go:17]: Actually, this is not that important. And if you deem that this change is necessary, then you can include it in a separate PR.

**Review by arunpoudel — APPROVED** [review]

File: middleware/database/database.go (hunk 1 above)
Comment (bendiknesbo) [reply middleware/database/database.go:17]: A remnant from a previous iteration :) 
I'll create a separate PR for improving it, but this func wont be in the hot-loop, so it won't matter much :)

**Pull request merged**